LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,337 @@
+                        CorrectionRangeOverridesRangeItem(
+                            preMealTargetRange: self.viewModel.therapySettings.preMealTargetRange,
+                            workoutTargetRange: self.viewModel.therapySettings.workoutTargetRange,
+                            unit: self.unit,
+                            preset: preset,
+                            correctionRangeScheduleRange: self.viewModel.therapySettings.glucoseTargetRangeSchedule!.scheduleRange()
+                        )
+                    }
+                } else {
+                    EmptyView()
+                }
+            }
+        }
+    }
+    
+}
+
+struct ScheduledRangeItem: View {
+    let time: TimeInterval
+    let range: DoubleRange
+    let unit: HKUnit
+    let guardrail: HKQuantityGuardrail
+    
+    public var body: some View {
+        ScheduleItemView(time: time,
+                         isEditing: .constant(false),
+                         valueContent: {
+                            GuardrailConstrainedQuantityRangeView(range: range.quantityRange(for: unit), unit: unit, guardrail: guardrail, isEditing: false)
+                         },
+                         expandedContent: { EmptyView() })
+    }
+}
+
+struct CorrectionRangeOverridesRangeItem: View {
+    let preMealTargetRange: DoubleRange?
+    let workoutTargetRange: DoubleRange?
+    let unit: HKUnit
+    let preset: CorrectionRangeOverrides.Preset
+    let correctionRangeScheduleRange: ClosedRange<HKQuantity>
+    
+    public var body: some View {
+        CorrectionRangeOverridesExpandableSetting(
+            isEditing: .constant(false),
+            value: .constant(CorrectionRangeOverrides(
+                preMeal: preMealTargetRange,
+                workout: workoutTargetRange,
+                unit: unit
+            )),
+            preset: preset,
+            unit: unit,
+            correctionRangeScheduleRange: correctionRangeScheduleRange,
+            expandedContent: { EmptyView() })
+    }
+}
+
+struct SectionHeaderWithEdit: View {
+    @Binding var isEditing: Bool
+    @Binding var editButtonDesignVersion: TherapySettingsView.DesignVersion
+    let title: String
+    let editAction: () -> Void
+
+    public var body: some View {
+        HStack(alignment: .firstTextBaseline) {
+            SectionHeader(label: title)
+            if isEditing && editButtonDesignVersion == .C {
+                Spacer()
+                Button(action: editAction) {
+                    Text("Edit")
+                        .font(.subheadline)
+
+                }
+                .disabled(!isEditing)
+            }
+        }
+    }
+}
+
+// Note: I didn't call this "EditableSection" because it doesn't actually make the section editable,
+// it just optionally provides a link to go to an editor screen.
+struct SectionWithEdit<Content, Footer>: View where Content: View, Footer: View {
```

Comment:
[nit] Since this `struct` provides a navigation link, could this be named `SectionWithNavigation`? For me, when I see it, I think that it's a section with an edit button but I don't automatically think about navigation

Reply:
The thing is, it doesn't _actually_ navigate.  It just calls out to a delegate method called `gotoEdit`.  Maybe that's just nuance...but all the other `Navigate` things in SwiftUI _actually_ navigate.  WDYT?

Reply:
Ahh, I didn't notice it was a delegate method call. Is there a reason for doing that vs a nav link? Is it so we can save the setting more easily?

Reply:
Two reasons:
1) It isn't necessarily the case that we will be using a `NavigationLink` because we're still ironing out the design.
2) With this view "living" in two worlds (onboarding & settings), I wanted to delegate the navigation responsibility to an outer class to deal with navigation, possibly in different ways for the two different situations.

Reply:
From an onboarding perspective, the navigation should be fairly similar if not exactly the same. Since this uses `TherapySettings` to back the screen, I'd be fine with using SwiftUI's standard API and making changes to the view model, if needed, to fully support the onboarding.